lib/middleware/request_tracker.rb

```diff
@@ -406,7 +406,11 @@ def call(env)
 
   def log_later(data)
     Scheduler::Defer.later("Track view") do
-      self.class.log_request(data) unless Discourse.pg_readonly_mode?
+      begin
+        self.class.log_request(data) unless Discourse.pg_readonly_mode?
+      rescue ActiveRecord::ReadOnlyError
+        # Just noop if ActiveRecord is preventing writes
```

Comment:
Should we log anything here to help with troubleshooting? Presumably a write during a GET request is some kind of logical error.

Reply:
Actually, I just realised my description in the PR is inaccurate. The GET request queues a "job" in the defer queue and that job happens to write to the DB. In this particular case, we have to give up if we can't write as there is not concept of retries in the defer queue at the moment.